Parsers/ASimRegistryEvent/Parsers/vimRegistryEventMicrosoftWindowsEvent.yaml

```diff
@@ -1,153 +1,390 @@
+            ,
+        "%%1883", "REG_QWORD"
+    ];
+        union isfuzzy=false
+            (
             WindowsEvent
-            | where EventID == 4663
-                and EventData.ObjectType == "Key"
-            | extend AccessMask = tostring(EventData.AccessMask)
-                , Type = "WindowsEvent"
+            | where not(disabled)
+            | where (isnull(starttime) or TimeGenerated >= starttime) 
+                and (isnull(endtime) or TimeGenerated <= endtime)
+            | where EventID == 4663 and EventData.ObjectType == "Key"
+            | where (array_length(actorusername_has_any) == 0 or (EventData.SubjectDomainName has_any (actorusername_has_any)) or (EventData.SubjectUserName has_any (actorusername_has_any)) or (strcat(EventData.SubjectDomainName, '\\', EventData.SubjectUserName) has_any (actorusername_has_any))) and
+                (array_length(registryvalue_has_any) == 0) and 
+                (array_length(registrydata_has_any) == 0) and
+                (array_length(dvchostname_has_any) == 0 or Computer has_any (dvchostname_has_any))
+            | extend
+                AccessMask = tostring(EventData.AccessMask)
+                ,
+                Type = "WindowsEvent"
             | lookup Event4663TypeLookup on AccessMask
+            | extend EventType = iif(isempty(EventType), "Other", EventType)
+            | where (array_length(eventtype_in) == 0 or EventType in~ (eventtype_in))
             | invoke ASIM_ParseWindowsEvents()
-            | project TimeGenerated, Computer, EventID, EventType, ActorUsername, ActorDomainName, ActorUserId, ActorSessionId, ActingProcessName, ActingProcessId, RegistryKey, _ResourceId, Type
-        ), (
+            | where (array_length(registrykey_has_any) == 0 or RegistryKey has_any (registrykey_has_any))
+            | project
+                TimeGenerated,
+                Computer,
+                EventID,
+                EventType,
+                ActorUsername,
+                ActorDomainName,
+                ActorUserId,
+                ActorSessionId,
+                ActingProcessName,
+                ActingProcessId,
+                RegistryKey,
+                _ResourceId,
+                Type
+            ),
+            (
             SecurityEvent
-            | where EventID == 4663
-                and ObjectType == "Key"
+            | where not(disabled)
+            | where (isnull(starttime) or TimeGenerated >= starttime) 
+                and (isnull(endtime) or TimeGenerated <= endtime)
+            | where EventID == 4663 and ObjectType == "Key"
+            | where (array_length(actorusername_has_any) == 0 or (SubjectDomainName has_any (actorusername_has_any)) or (SubjectUserName has_any (actorusername_has_any)) or (strcat(SubjectDomainName, '\\', SubjectUserName) has_any (actorusername_has_any))) and
+                (array_length(registryvalue_has_any) == 0) and 
+                (array_length(registrydata_has_any) == 0) and
```

Comment:
Any reason why registryvalue_has_any and registrydata_has_any does not considered for filteration

Reply:
incase of 4663 eventid, registryvalue and registrydata mapping is not present in the parser. I checked the data, the value for these two fields is not coming for 4663